spacy/kb/candidate.pxd

```diff
@@ -0,0 +1,12 @@
+from .kb cimport KnowledgeBase
+from libcpp.vector cimport vector
+from ..typedefs cimport hash_t
+
+# Object used by the Entity Linker that summarizes one entity-alias candidate combination.
```

Comment:
I find the term `alias` very misleading. If we're doing major refactoring, perhaps we could consider changing it to `mention`?

Reply:
That's a good point, but it would be breaking. Other attributes in `Candidate` should IMO also be renamed or dropped, since they aren't independent from the implementation of `KnowledgeBase`. E.g. `entity_freq`, or the fact that depending on the implementation there is no need for hashes - the minimal set of attributes would IMO be just an `entity_id` and an abstract property for returning some identifiable version of the mention.

We could rename `alias_hash` to `mention_hash` and add a property `alias_hash` returning the value of `mention_hash` to maintain backwards compatibility. At the same time it might be better to leave things as they are - i.e. slightly misleading - and do a bigger refactoring only with v4 (which is what I'd opt for).

What do you think, @svlandeg?

Reply:
IIRC, in the original code `alias` and `mention` are used for two slightly different things. A mention is an actual `Span` in an actual `Doc`, and an alias is a synonym or term associated to a certain identifier in the knowledge base / ontology.

The confusion then was introduced because the built-in look-up function only uses the textual string of the mention to retrieve alias-ID pairs from the KB that could be relevant for that mention, at which point "alias" and "mention" became almost interchangeable because both refered to the textual string only and even verbatim (no fuzzy matching). But this is only true for the built-in look-up implementation. In the more general case, I agree that a `Candidate` should in fact link a `mention` to a likely entity, and whether this matching happens through an alias or an embedding similarity or whatever should be irrelevant for the `entity_linker` (because it's the candidate generator's job).

As Raphael points out, all of this refactoring/renaming should be kept for `v4` though. (are you keeping track of these things for the next PR? ;-))

Reply:
I do now :wink: @richardpaulhudson Closing this conversation since we decided not to rename. If you feel there's a need for further discussion, feel free to reopen.